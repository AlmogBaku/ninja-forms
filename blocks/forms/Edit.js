import React from 'react';

export default  function Edit() {
    return <React.Fragment/>
}