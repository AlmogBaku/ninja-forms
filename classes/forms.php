<?php
/**
 * Handles adding and removing forms.
 *
 * @package     Ninja Forms
 * @subpackage  Classes/Form
 * @copyright   Copyright (c) 2014, WPNINJAS
 * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
 * @since       2.9
*/

class NF_Forms {

	/**
	 * Get things started
	 * 
	 * @access public
	 * @since 2.9
	 * @return void
	 */
	public function __construct() {
		
	}

	/**
	 * Get all forms
	 * 
	 * @access public
	 * @since 2.9
	 * @return array $forms
	 */
	public function get_all( $debug = false, $show_new = false ) {
		$forms = nf_get_objects_by_type( 'form' );

		$tmp_array = array();
		foreach ( $forms as $form ) {
			$form_id = $form['id'];

			$status = Ninja_Forms()->form( $form_id )->get_setting( 'status' );
			if ( ( $status == 'new' && $show_new ) || $status != 'new' ) {
				$title = Ninja_Forms()->form( $form_id )->get_setting( 'form_title' );
				if ( strpos( $title, '_' ) === 0 ) {
					if ( $debug )
						$tmp_array[] = $form_id;
				} else {
					$tmp_array[] = $form_id;
				}
			}
		}
		return $tmp_array;
	}
}