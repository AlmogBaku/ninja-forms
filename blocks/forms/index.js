import Edit from './Edit';
import  block from './block';
import { __ } from '@wordpress/i18n';
import { registerBlockType } from '@wordpress/blocks';

registerBlockType('ninja-forms/form',{
    ...block,
    title: __('Ninja Form', 'ninja-forms'),
    edit: ({attributes,setAttributes}) => {

        return <Edit/>
    },
    save: () => null
});