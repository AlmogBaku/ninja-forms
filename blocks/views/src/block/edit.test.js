import { Edit } from './edit';
import {
   render, //test renderer
   cleanup, //resets the JSDOM
   fireEvent, //fires events on nodes,
} from "@testing-library/react";

import forms from "../data/forms";

describe("Editor component", () => {
    afterEach(cleanup);

    it("matches snapshot", () => {
      const attributes = {
            formId: 1
    };
      const setAttributes = jest.fn();

      const {container} = render(
        <Edit
          {...{
            attributes,
            setAttributes,
            forms: {},
            fields: forms[1].fields,
            submissions: forms[1].submissions,
            clientId: "random-id",
            className: "wp-blocks-whatever"
          }}
        />
      )
      
      expect(container).toMatchSnapshot();
    });

});