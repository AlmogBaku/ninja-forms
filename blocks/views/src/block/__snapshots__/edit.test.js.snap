// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor component matches snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm cYunTa"
  >
    <table
      role="table"
    >
      <thead />
      <tbody
        role="rowgroup"
      />
    </table>
    <div
      style="display: flex; justify-content: space-between;"
    >
      <div />
    </div>
  </div>
</div>
`;
